File: LeetCode/RemoveElement.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int k = nums.size();
        int count = 0;
        for(int i=0;i<k;i++){
            if(val != nums[i]){
                nums[count++] = nums[i];
            }
        }
    
        return count;
    }
};

File: LeetCode/RemoveElement.php
<?php
class Solution {

/**
 * @param Integer[] $nums
 * @param Integer $val
 * @return Integer
 */
function removeElement(&$nums, $val) {
    $k = count($nums);
    $ct = 0;
    for ($i=0;$i<$k;$i++){
        if ($nums[$i] != $val){
            $nums[$ct] = $nums[$i];
            $ct++;
        }
    }
    return $ct;
}
}